### PR TITLE
Compile http4s-client and http4s-server on 3.0.0-M2

### PR DESCRIPTION
**Pull request by rossabaker**
These are the necessary changes I wish weren't so.

### CODE REVIEW

**Comment by rossabaker** [comment]
http4s-server doesn't yet include tests, which reveal new pain:

```
[error] -- [E040] Syntax Error: /home/ross/src/http4s/server/src/test/scala/org/http4s/server/ContextRouterSuite.scala:27:68 
[error] 27 |  val numbers = ContextRoutes.of[Unit, IO] { case GET -> Root / "1" as _ =>
[error]    |                                                                    ^^
[error]    |                                     '=>' expected, but identifier found
```

**Comment by rossabaker** [comment]
Thanks to @nafg and @ChristopherDavenport for helping me on the server change.

**Comment by rossabaker** [comment]
Uh-oh, we broke FollowRedirect with this.

File: core/src/main/scala/org/http4s/DecodeResult.scala [hunk 1]
```diff
@@ -30,9 +30,15 @@ object DecodeResult {
   def success[F[_], A](a: A)(implicit F: Applicative[F]): DecodeResult[F, A] =
     EitherT(F.pure(Right(a)))
 
+  def successT[F[_], A](a: A)(implicit F: Applicative[F]): DecodeResult[F, A] =
```
Comment (rossabaker) [inline core/src/main/scala/org/http4s/DecodeResult.scala:33]: See #4268.  If we proceed, we should start a deprecation cycle in 0.21 and move these down there.

File: client/src/main/scala/org/http4s/client/oauth1/oauth1.scala [hunk 2]
```diff
@@ -78,7 +78,10 @@ package object oauth1 {
       verifier: Option[Verifier] = None
   )(implicit F: MonadThrow[F], W: EntityDecoder[F, UrlForm]): F[Request[F]] =
     for {
-      (req, params) <- getUserParams(req)
+      reqParams <- getUserParams(req)
+      // Working around lack of withFilter
```
Comment (ChristopherDavenport) [inline client/src/main/scala/org/http4s/client/oauth1/oauth1.scala:82]: `(req, params) = reqParams`

Comment (rossabaker) [reply client/src/main/scala/org/http4s/client/oauth1/oauth1.scala:82]: I miss better-monadic-for already.

**Review by hamnis — APPROVED** [review]

**Pull request merged**